data.js

```diff
@@ -0,0 +1,182 @@
+        // TODO: would be great to prune 311 data to the last 30 days, like the police data
+        "311": {
+            id: "40776043-ad00-40f5-9dc8-1fde865ff571",
+            primaryFiltering: "WHERE \"NEIGHBORHOOD\" LIKE '%Oakland' ORDER BY \"CREATED_ON\" DESC",
+            latLong: ["Y", "X"],
+            icon: iconTypes.CITY_311_ICON,
+
+            title: (record) => record["REQUEST_TYPE"],
+            popup: (record) => `
+              <strong>${record["DEPARTMENT"]}</strong>
+              <br> ${record["REQUEST_TYPE"]}`,
+
+            processRecord: (record) => {
+                // Collect time of incident from the record
+                record.incidentYear = parseInt(record.CREATED_ON.substring(0,4));
+                record.incidentMonth = parseInt(record.CREATED_ON.substring(5,8));
+                record.incidentDay = parseInt(record.CREATED_ON.substring(8,10));
+            }
+        },
+
+        // Calls from the library db
+        "Library": {
+            id: "2ba0788a-2f35-43aa-a47c-89c75f55cf9d",
+            primaryFiltering: "WHERE \"Name\" LIKE '%OAKLAND%'",
+            latLong: ["Lat", "Lon"],
+            icon: iconTypes.LIBRARY_ICON,
+
+            title: (record) => record["Name"],
+            popup: (record) => `
+              <strong>${record.Name}</strong>
+              <br> Address: ${record.Address}
+              <br> Phone: ${record.Phone}
+              <br> Monday: ${record.MoOpen.substring(0, 5)} - ${record.MoClose.substring(0, 5)}
+              <br> Tuesday: ${record.TuOpen.substring(0, 5)} - ${record.TuClose.substring(0, 5)}
+              <br> Wednesday: ${record.WeOpen.substring(0, 5)} - ${record.WeClose.substring(0, 5)}
+              <br> Thursday: ${record.ThOpen.substring(0, 5)} - ${record.ThClose.substring(0, 5)}
+              <br> Friday: ${record.FrOpen.substring(0, 5)} - ${record.FrClose.substring(0, 5)}
+              <br> Saturday: ${record.SaOpen.substring(0, 5)} - ${record.SaClose.substring(0, 5)}
+              <br> Sunday: ${record.SuOpen.substring(0, 5)} - ${record.SuClose.substring(0, 5)}
+              `,
+
+        },
+        "Non-Traffic Violation": {
+            id: "6b11e87d-1216-463d-bbd3-37460e539d86",
+            primaryFiltering: "Where \"NEIGHBORHOOD\" LIKE '%Oakland'",
+            latLong: ["Y", "X"],
+            icon: iconTypes.NON_TRAFFIC_VIOLATION,
+
+            title: (record) => record["OFFENSES"],
+            popup: (record) => record["OFFENSES"],
+
+            processRecord: (record) => {
+                record.incidentYear = parseInt(record.CITEDTIME.substring(0,4));
+                record.incidentMonth = parseInt(record.CITEDTIME.substring(5,8));
+                record.incidentDay = parseInt(record.CITEDTIME.substring(8,10));
+            }
+        }
+    };
+
+    const WPRDC_QUERY_PREFIX = "SELECT * FROM \"";
+    const WPRDC_QUERY_SUFFIX = "\" ";
+
+    const dataReadyEvent = new CustomEvent("dataready", {
```

Comment:
Really interesting idea. I think you're right that the traditional solution would be to use the `window` object being loaded on to both IIFEs, loading constants like these like `window.detail = { //etc }`, then conditionally finding them in the next one. On the other, I completely buy your argument; this is the best way for us.